HLTrigger/HLTcore/interface/HLTConfigProvider.h

```diff
@@ -183,7 +184,10 @@ class HLTConfigProvider {
   /// Number of HLT prescale sets
   unsigned int prescaleSize() const { return hltConfigData_->prescaleSize(); }
   /// HLT prescale value in specific prescale set for a specific trigger path
-  unsigned int prescaleValue(unsigned int set, const std::string& trigger) const {
+  template<typename T = unsigned int>
+  T prescaleValue(unsigned int set, const std::string& trigger) const {
+    //limit to only 4 allowed types
+    static_assert(std::is_same_v<T, unsigned int> or std::is_same_v<T,FractionalPrescale> or std::is_same_v<T,int> or std::is_same_v<T,double>, "Please use prescaleValue<unsigned int>, prescaleValue<int>, prescaleValue<double>, or prescaleValue<FractionalPrescale>,\n note int and unsigned int will be depreated soon");
```

Comment:
since the old return type was `unsigned int`, why do we add `int` as well ?

Reply:
because prescaleValues was int,int  and therefore it seemed easier to allow this as a type so that prescaleValues could just pass the type straight through to prescaleValue

Note we'll be dropping these soon anyways.